composer/core/state.py

```diff
@@ -217,6 +231,9 @@ def __init__(
         self.set_dataloader(dataloader, dataloader_label, dataloader_len)
         self.max_duration = max_duration
 
+        self.train_dataloader = train_dataloader
+        self.evaluators = list(ensure_tuple(evaluators))
```

Comment:
Similar to how the optimizers and schedulers are handled, can we use a private getter / setter that does an in-place modification of the list? This is mainly to ensure that any code like `evaluators = state.evaluators` always points to the latest evaluators.